src/ascent/runtimes/flow_filters/ascent_runtime_vtkh_filters.cpp

```diff
@@ -380,6 +389,238 @@ vtkh::Render parse_render(const conduit::Node &render_node,
   return render;
 }
 
+class CinemaManager
+{
+protected:  
+  std::vector<vtkm::rendering::Camera> m_cameras;
+  std::vector<std::string>             m_image_names;
+  std::vector<float>                   m_phi_values;
+  std::vector<float>                   m_theta_values;
+  std::vector<float>                   m_times;
+
+  vtkm::Bounds                         m_bounds;
+  const int                            m_phi;
+  const int                            m_theta;
+  std::string                          m_image_name;
+  std::string                          m_current_path;
+  float                                m_time;
+public:
+  CinemaManager(vtkm::Bounds bounds, 
+                const int phi, 
+                const int theta, 
+                const std::string image_name)
+    : m_bounds(bounds),
+      m_phi(phi),
+      m_theta(theta),
+      m_image_name(image_name),
+      m_time(0.f)
+  {
+    this->create_cinema_cameras(bounds);
+  }
+  
+  CinemaManager() 
+    : m_phi(0),
+      m_theta(0)
+  {
+    ASCENT_ERROR("Cannot create un-initialized CinemaManger"); 
+  }
+
+  void add_time_step()
+  {
+    m_times.push_back(m_time);
+    m_time += 1.f;
+
+    // add top level dir
+    string output_path = "cinema_databases";
+    
+    if(!conduit::utils::is_directory(output_path))
+    {
+        conduit::utils::create_directory(output_path);
```

Comment:
for MPI case, do we need to guard this to only occur on rank 0?

We could create our own ascent util (which uses the conduit one) that handles the MPI vs non MPI case transparently.

Reply:
You are right. We don't need all ranks trying to create the same directory.

Reply:
Ok. Fixed. I just grabbed the mpi comm and checked rank. I really don't see us have many other use cases for creating directories in filters.

Reply:
sounds good, make sure to push :-)